src/routes/install.ts

```diff
@@ -6,122 +6,122 @@
  * (Department of Information and Computing Sciences)
  */
 
-import { Request, Response } from "express";
 import { Context } from "../context";
+import { SessionData } from "../types";
+import { z } from "zod";
 
-/**
- * The request type for the installation of an addon.
- */
-interface InstallRequest extends Request {
-  // The validation middleware will ensure that the body contains the required
-  // fields, so we can safely assume that they are present and don't need to
-  // mark them as optional.
-  body: {
-    userId: string;
-    addonId: string;
-  };
+////////////////////////////////////////////////////////////////////////////////
+
+interface InstallRequest {
+  addonID: string;
 }
 
-/**
- * Handles the installation of an addon for a user.
- *
- * @param ctx - The context instance.
- * @returns An async function that handles the installation request.
- */
-export const installRoute =
-  (ctx: Context) => async (req: InstallRequest, res: Response) => {
-    const { userId, addonId } = req.body;
+const installSchema = z.object({
+  addonID: z.string()
+});
+
+interface InstallResponseData {}
```

Comment:
Returning a promise of an empty interface means that any object is allowed to be returned. Ideally, we don't return at all or clarify more clearly what we return. Anyhow, we need to communicate this with the frontend people to see what they need for the install button.